compiler/ccgutils.nim

```diff
@@ -53,7 +53,9 @@ proc hashString*(conf: ConfigRef; s: string): BiggestInt =
     a = a + (a shl 3)
     a = a xor (a shr 11)
     a = a + (a shl 15)
+    {.push warnings: off.}
     result = cast[Hash](a)
```

Comment:
cast will generally fill the extra bytes with random garbage when creating a bigger type - this doesn't smell safe in a hash function

Reply:
ie, it doesn't smell safe anywhere really, except when you're about to overwrite the rest of the bytes with other data - the better fix here would actually be to not use a `cast` and instead use `BiggestInt(a)`